hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -70,22 +70,18 @@ protected AbstractMessageTask(ClientMessage clientMessage, Node node, Connection
         this.connection = connection;
         this.clientEngine = node.clientEngine;
         this.endpointManager = clientEngine.getEndpointManager();
-        this.endpoint = getEndpoint();
+        this.endpoint = initEndpoint();
     }
 
     @SuppressWarnings("unchecked")
     public <S> S getService(String serviceName) {
         return (S) node.nodeEngine.getService(serviceName);
     }
 
-    protected ClientEndpoint getEndpoint() {
+    protected ClientEndpoint initEndpoint() {
```

Comment:
This is problematic (or at least ugly). It is very dangerous to call an overridable method from a constructor, because if it gets overridden in a subclass, then overridden method can be called from the super class constructor, without the subclass constructor completed.

See the following example:

```

public class Main {

    public static void main(String[] args){
        Bar bar = new Bar();
     }
}


class Foo{
    Foo(){
        foo();
    }

    public void foo(){}
}

class Bar extends Foo{
    final int a;

    Bar(){
        a=10;
    }

   public void foo(){
        System.out.println(a);
    }
}
```

It will print 0.